src/R/Components/Impl/PackageManager/Implementation/RestartManager.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Microsoft.R.Components.PackageManager.Implementation {
+    internal class RestartManager {
+        /// <summary>
+        /// Given a list of filenames with aboslute path, returns list of process currently locking those files.
+        /// </summary>
+        /// <param name="filePaths">Filenames with absolute path.</param>
+        /// <returns>List of processes locking the files. null if it encounters any error.</returns>
+        public static IList<Process> GetProcessesUsingFiles(string[] filePaths) {
```

Comment:
I would make this `IEnumerable<Process>`, and use `yield return` inside to fetch them on demand. The caller can then do `ToList` or not, as needed.